autotx/AutoTx.py

```diff
@@ -76,93 +97,29 @@ def run(self, prompt: str, non_interactive: bool) -> None:
-                    """
-                ),
-                llm_config=self.get_llm_config(),
-                human_input_mode="NEVER",
-                code_execution_config=False,
-            )
+            helper_agents: list[AutogenAgent] = [
+                user_proxy_agent,
+                verifier.build(self.get_llm_config),
+                clarifier_agent
+            ]
 
-            # Print all messages sent form the verifier to the group chat manager,
-            # as they tend to contain valuable information
-            def verifier_send_message_hook(
-                sender: AssistantAgent,
-                message: Union[Dict[str, Any], str],
-                recipient: AutogenAgent,
-                silent: bool
-            ) -> Union[Dict[str, Any], str]:
-                if recipient.name == "chat_manager" and message != "TERMINATE":
-                    if isinstance(message, str):
-                        cprint(message, "light_yellow")
-                    elif message["content"] != None:
-                        cprint(message["content"], "light_yellow")
-                return message
-
-            verifier_agent.register_hook(
-                "process_message_before_send",
-                verifier_send_message_hook
-            )
+            autogen_agents = [agent.build_autogen_agent(self, user_proxy_agent, self.get_llm_config()) for agent in self.agents]
 
-            autogen_agents = [agent.build_autogen_agent(self, user_proxy, self.get_llm_config()) for agent in self.agents]
-
-            groupchat = GroupChat(
-                agents=autogen_agents + [user_proxy, verifier_agent], 
-                messages=[], 
-                max_round=20,
-                select_speaker_prompt_template = (
-                    """
-                    Read the above conversation. Then select the next role from {agentlist} to play. Only return the role and NOTHING else.
-                    If agents are trying to communicate with the user, or requesting approval, return the 'user_proxy' role.
-                    """
-                )
-            )
-            manager = GroupChatManager(groupchat=groupchat, llm_config=self.get_llm_config())
+            manager_agent = manager.build(autogen_agents + helper_agents, self.get_llm_config)
 
-            chat = user_proxy.initiate_chat(manager, message=dedent(
-                f"""
-                    My goal is: {prompt}
-                    Advisor reworded: {goal}
-                """
-            ))
+            chat = user_proxy_agent.initiate_chat(manager_agent, message=prompt, summary_method=summary_method)
 
             if "ERROR:" in chat.summary:
                 error_message = chat.summary.replace("ERROR: ", "").replace("\n", "")
                 cprint(error_message, "red")
             else:
                 cprint(chat.summary.replace("\n", ""), "green")
```

Comment:
i think we can safely remove this
```suggestion
```
otherwise we get `Goal not supported` twice (and in different colors)
![image](https://github.com/polywrap/AutoTx/assets/21031138/d76da4d9-f1f4-4028-8a64-46c5fdd4e0ef)

Reply:
Isn't this needed for other cases when there is an error?

Reply:
the verifier is not throwing error anymore, so nope